tempesta_fw/cache.c

```diff
@@ -308,6 +308,9 @@ tfw_cache_strcpy_eol(char **p, TdbVRec **trec,
 
 	BUG_ON(TFW_STR_DUP(src));
 
+	if (!src->len)
+		goto out;
```

Comment:
Can't that that it makes sense here. The thread of execution is finished here, so it's best to return right here instead of mixing it with the other thread of execution. The two threads of execution don't have any common code.